Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -396,8 +396,21 @@ private void OnStateChanged(object requestor, CancellationToken token, ParserSta
             var handler = StateChanged;
             if (handler != null && !token.IsCancellationRequested)
             {
-                var args = new ParserStateEventArgs(state, oldStatus, token);
-                handler.Invoke(requestor, args);
+                try
+                {
+                    var args = new ParserStateEventArgs(state, oldStatus, token);
+                    handler.Invoke(requestor, args);
+                }
+                catch (OperationCanceledException cancellation)
```

Comment:
Complete nitpick: you do not need to specify the name of the exception, since it is not used.